fvm/errors/txVerifier.go

```diff
@@ -6,129 +6,23 @@ import (
-// - ReferenceBlock.Height - CurrentBlock.Height < Expiry Limit (Transaction is Expired)
-type ExpiredTransactionError struct {
-	refHeight, finalHeight uint64
-}
-
-// NewExpiredTransactionError constructs a new ExpiredTransactionError
-func NewExpiredTransactionError(refHeight, finalHeight uint64) *ExpiredTransactionError {
-	return &ExpiredTransactionError{refHeight: refHeight, finalHeight: finalHeight}
-}
-
-func (e ExpiredTransactionError) Error() string {
-	return fmt.Sprintf("%s transaction is expired: ref_height=%d final_height=%d", e.Code().String(), e.refHeight, e.finalHeight)
-}
-
-// Code returns the error code for this error type
-func (e ExpiredTransactionError) Code() ErrorCode {
-	return ErrCodeInvalidReferenceBlockError
-}
-
-// InvalidScriptError indicates that a transaction contains an invalid Cadence script.
-// this error is the result of failure in any of the following conditions:
-// - script is empty
-// - script can not be parsed by the cadence parser
-// - comment-only script, len(program.Declarations) == 0
-type InvalidScriptError struct {
-	err error
-}
-
-// NewInvalidScriptError constructs a new InvalidScriptError
-func NewInvalidScriptError(err error) *InvalidScriptError {
-	return &InvalidScriptError{err: err}
-}
-
-func (e InvalidScriptError) Error() string {
-	return fmt.Sprintf("%s failed to parse transaction Cadence script: %s", e.Code().String(), e.err.Error())
-}
-
-// Code returns the error code for this error type
-func (e InvalidScriptError) Code() ErrorCode {
-	return ErrCodeInvalidScriptError
-}
-
-// Unwrap unwraps the error
-func (e InvalidScriptError) Unwrap() error {
-	return e.err
-}
-
-// InvalidGasLimitError indicates that a transaction specifies a gas limit that exceeds the maximum allowed by the network.
-type InvalidGasLimitError struct {
-	actual  uint64
-	maximum uint64
-}
-
-// NewInvalidGasLimitError constructs a new InvalidGasLimitError
-func NewInvalidGasLimitError(actual, maximum uint64) *InvalidGasLimitError {
-	return &InvalidGasLimitError{actual: actual, maximum: maximum}
-}
-
-func (e InvalidGasLimitError) Error() string {
-	return fmt.Sprintf("%s transaction gas limit (%d) exceeds the maximum gas limit (%d)", e.Code().String(), e.actual, e.maximum)
-}
-
-// Code returns the error code for this error type
-func (e InvalidGasLimitError) Code() ErrorCode {
-	return ErrCodeInvalidGasLimitError
-}
-
 // InvalidProposalSignatureError indicates that no valid signature is provided for the proposal key.
 type InvalidProposalSignatureError struct {
+	errorWrapper
+
 	address  flow.Address
 	keyIndex uint64
-	err      error
 }
 
 // NewInvalidProposalSignatureError constructs a new InvalidProposalSignatureError
 func NewInvalidProposalSignatureError(address flow.Address, keyIndex uint64, err error) *InvalidProposalSignatureError {
-	return &InvalidProposalSignatureError{address: address, keyIndex: keyIndex, err: err}
+	return &InvalidProposalSignatureError{
```

Comment:
do you want to change this to non-pointer as well (same below)?

Reply:
missed that one, thanks!